matrix_registration/templates/register.html

```diff
@@ -48,9 +48,9 @@ <h1>{{ translations.server_registration }}</h1>
       <section>
         <form id="registration" action="{{ base_url }}/register" method="post">
           <div class="group">
-            <input id="username" name="username" type="text" placeholder=" " required
-              pattern="^@?[a-zA-Z_\-=\.\/0-9]+(:{{ server_name|replace('.', '\.') }})?$" required minlength="1"
-              maxlength="200">
+            <input id="username" name="username" type="text" placeholder=" "
+              pattern="^@?(?=({{ uname_regex }}))[a-zA-Z_\-=\.\/0-9]+(:{{ server_name|replace('.', '\.') }})?$" 
```

Comment:
The brakets around {{uname_regex}} aren't required but serve as a protection against missconfigured configs

Reply:
nvm works just as well without - increasing regex speed by ~10%